File: CS/WindowsFormsApplication154/CustomTextEditViewInfo.cs
using DevExpress.Utils.Drawing;
using DevExpress.XtraEditors.Repository;
using DevExpress.XtraEditors.ViewInfo;

namespace WindowsFormsApplication154 {
    public class CustomTextEditViewInfo : TextEditViewInfo {
        public CustomTextEditViewInfo(RepositoryItem item) : base(item) {
        }
        protected override BorderPainter GetBorderPainterCore() {
            if (this.BorderStyle == DevExpress.XtraEditors.Controls.BorderStyles.UltraFlat) {
                return new CustomUltraFlatBorderPainter();
            }
            return base.GetBorderPainterCore();
        }
    }
}

File: VB/WindowsFormsApplication154/CustomTextEditViewInfo.vb
Imports DevExpress.Utils.Drawing
Imports DevExpress.XtraEditors.Repository
Imports DevExpress.XtraEditors.ViewInfo

Namespace WindowsFormsApplication154
    Public Class CustomTextEditViewInfo
        Inherits TextEditViewInfo

        Public Sub New(ByVal item As RepositoryItem)
            MyBase.New(item)
        End Sub
        Protected Overrides Function GetBorderPainterCore() As BorderPainter
            If Me.BorderStyle = DevExpress.XtraEditors.Controls.BorderStyles.UltraFlat Then
                Return New CustomUltraFlatBorderPainter()
            End If
            Return MyBase.GetBorderPainterCore()
        End Function
    End Class
End Namespace
